kernel/syscall.c

```diff
@@ -3126,6 +3126,57 @@ long myst_syscall_mbind(
     return ret;
 }
 
+long myst_syscall_get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    unsigned long rounded_maxnode = 0;
+    int ROUND_UP_VAL = sizeof(unsigned long) * 8;
+    myst_round_up(maxnode, ROUND_UP_VAL, &rounded_maxnode);
+
+    unsigned long* rounded_nodemask;
+    // allocate higher buffer
+    if (rounded_maxnode > maxnode)
+    {
+        rounded_nodemask = malloc(rounded_maxnode / CHAR_BIT);
+    }
+    else
+    {
+        rounded_nodemask = nodemask;
+    }
+
+    long params[6] = {(long)mode,
+                      (long)rounded_nodemask,
+                      (long)rounded_maxnode,
+                      (long)addr,
+                      (long)flags};
+    long ret = myst_tcall(SYS_get_mempolicy, params);
+
+    // As Mystikos does not support mbind or set_mempolicy, we override nodemask
+    // returned by the target to indicate the thread is not allowed to specify a
+    // policy on any node in mbind or set_mempolicy when
+    // flags = MPOL_F_MEMS_ALLOWED
+    if (flags == 1 << 2)
```

Comment:
If `ret` indicate error, skip any operation on nodemask.

Reply:
Good catch, adding in next commit